### PR TITLE
zc - fix bug for ability to navigate to unjoined commons

### LINKED ISSUE #6: BUG: Disallow ability to navigate to unjoined commons
### Have you updated the title?

- [x] I have updated the issue's title.

### Summary

If a user is not currently part of a commons, they can still navigate to the URL for that commons and get a undefined page: 

Screenshot 2024-05-15 at 11 42 21 AM

### Expected Behavior

The user should be able to see some dialogue that indicates that they have not joined this commons yet, or they should be redirected to a page that tells them they do not have access.

### Current / Observed Behavior

This is the current behavior:

Screenshot 2024-05-15 at 11 44 03 AM

### Steps to Reproduce

1. Create a new commons (as an admin), but do not join the commons.
2. Navigate to the `https://happycows.dokku-XX.cs.ucsb.edu/play/{commonsId}` page
3. Observe the behavior

### Browser / Platform Info

_No response_

### PR DESCRIPTION
**Pull request by imnotmomo**
## Overview
In this PR, I fixed the bug that a user can navigate to an unjoined common (by directly modifying the common id in url) by adding a redirect page (changed App.js user role redirection and added a new access page) (so that the user is navigated to the existed 404 Not Found page)

## Screenshots
WeChatef81dd413d8f4a547f1bc24fbd2f917f
when accessing common 2 which I did not join
WeChat85001fbd962a97f9834b085e2aad1359
WeChat307333f019a09c85c7e745eb9d511b39

## Validation
1. Go to https://happycows-chendashi-dev.dokku-07.cs.ucsb.edu/
2. There should be two commons already exist
3. visit common1, and change url https://happycows-chendashi-dev.dokku-07.cs.ucsb.edu/play/2
4. you will be redirected to the 404 Not Found Page

## Tests
-  Frontend Test coverage (`npm run coverage`) 100%
-  Frontend Mutation tests (`npx stryker run`) 100% 

## Linked Issues
Closes #6

### CODE REVIEW

**Review by YixinXue — DISMISSED** [review]
LGTM

File: frontend/src/App.js
Comment (AnikaArora) [inline frontend/src/App.js:1]: remove the commented import statement

**Review by AnikaArora — CHANGES_REQUESTED** [review]
Please add a link to your dokku deployment for this branch so we can test out the changes ourselves.

File: frontend/src/App.js
Comment (imnotmomo) [reply frontend/src/App.js:1]: Fixed

**Review by YixinXue — APPROVED** [review]
lgtm

**Review by AnikaArora — APPROVED** [review]
LGTM!

**Pull request merged**